### PR TITLE
feat: Add sideloaded kit example

### PR DESCRIPTION
**Pull request by rmi22186**
## Instructions
1. PR target branch should be against development
2. PR title name should follow this format: https://github.com/mParticle/mparticle-workflows/blob/stable/.github/workflows/pr-title-check.yml
3. PR branch prefix should follow this format: https://github.com/mParticle/mparticle-workflows/blob/stable/.github/workflows/pr-branch-check-name.yml

## Summary
This PR includes a custom sideloaded kit that will fire on every mapped mParticle method.  The sideloaded kit was placed as a sibling of `higgs-shop-sample-app`.

## Testing Plan
- Launched Higgs app and checked the console to see if the sideloaded kit properly console.log'd mapped methods.  See example below:
image

## Master Issue
- Closes https://go.mparticle.com/work/SQDSDKS-5116

### CODE REVIEW

File: core-sdk-samples/mp-sideloaded-kit-example/src/initialization.js
Comment (alexs-mparticle) [inline core-sdk-samples/mp-sideloaded-kit-example/src/initialization.js:1]:
```suggestion
const initialization = {
```

File: core-sdk-samples/mp-sideloaded-kit-example/src/session-handler.js
Comment (alexs-mparticle) [inline core-sdk-samples/mp-sideloaded-kit-example/src/session-handler.js:1]:
```suggestion
const sessionHandler = {
```

File: core-sdk-samples/mp-sideloaded-kit-example/src/user-attribute-handler.js
Comment (alexs-mparticle) [inline core-sdk-samples/mp-sideloaded-kit-example/src/user-attribute-handler.js:17]: Can you run this through prettier/eslint? Some of your indenting looks off.

File: core-sdk-samples/higgs-shop-sample-app/src/layouts/App/index.tsx
Comment (alexs-mparticle) [inline core-sdk-samples/higgs-shop-sample-app/src/layouts/App/index.tsx:30]: I think you should leave this space. Makes things more readable.

File: core-sdk-samples/higgs-shop-sample-app/src/layouts/App/index.tsx
Comment (alexs-mparticle) [inline core-sdk-samples/higgs-shop-sample-app/src/layouts/App/index.tsx:6]: Nitpick: move this after `import mParticle` so we can group our stuff together.

**Review by alexs-mparticle — CHANGES_REQUESTED** [review]

File: core-sdk-samples/higgs-shop-sample-app/src/layouts/App/mpconfig.prod.json
Comment (alexs-mparticle) [inline core-sdk-samples/higgs-shop-sample-app/src/layouts/App/mpconfig.prod.json:1]: Why is this file being committed within the layouts?

File: core-sdk-samples/mp-sideloaded-kit-example/package.json
Comment (alexs-mparticle) [inline core-sdk-samples/mp-sideloaded-kit-example/package.json:1]: I thought you were going to move the kit into a sibling folder of `core-sdk-samples` rather than a child directory. Was I mistaken?

File: core-sdk-samples/mp-sideloaded-kit-example/dist/SampleSideLoadedKit-Kit.common.js
Comment (alexs-mparticle) [inline core-sdk-samples/mp-sideloaded-kit-example/dist/SampleSideLoadedKit-Kit.common.js:1]: Should we have a build task that produces these, rather than have them get committed?

File: core-sdk-samples/mp-sideloaded-kit-example/package.json
Comment (rmi22186) [reply core-sdk-samples/mp-sideloaded-kit-example/package.json:1]: accidental

File: core-sdk-samples/higgs-shop-sample-app/src/layouts/App/mpconfig.prod.json
Comment (rmi22186) [reply core-sdk-samples/higgs-shop-sample-app/src/layouts/App/mpconfig.prod.json:1]: accidental

**Comment by einsteinx2** [comment]
Seems like the sideloaded kit wasn't checked in, just the app changes?

**Comment by rmi22186** [comment]
> Seems like the sideloaded kit wasn't checked in, just the app changes?

@einsteinx2 - this was included initially but alex had some feedback about moving it, and it was dropped in one of the commits.  it has now been added back after testing.

**Review by einsteinx2 — DISMISSED** [review]
LGTM, I'll defer to Alex on final approval

**Comment by rmi22186** [comment]
This is blocked by a release for the web sdk to support the feature.  Once that has been released, we can merge this upon approval.

File: package-lock.json
Comment (alexs-mparticle) [inline package-lock.json:1]: Please delete this file

**Pull request merged**
